lib/internal/streams/legacy.js

```diff
@@ -2,16 +2,27 @@
 
 const {
   ObjectSetPrototypeOf,
+  SymbolAsyncIterator
 } = primordials;
 
 const EE = require('events');
 
+let createReadableStreamAsyncIterator;
+
 function Stream(opts) {
   EE.call(this, opts);
 }
 ObjectSetPrototypeOf(Stream.prototype, EE.prototype);
 ObjectSetPrototypeOf(Stream, EE);
 
+Stream.prototype[SymbolAsyncIterator] = function() {
+  if (createReadableStreamAsyncIterator === undefined) {
+    createReadableStreamAsyncIterator =
+      require('internal/streams/async_iterator');
+  }
+  return createReadableStreamAsyncIterator(this);
```

Comment:
We might want to be a bit smart here and check if the current stream can be iterated and provide a meaningful error. This has the unwanted consequeces of making `OutgoingMessage` async iterable.

Reply:
How about now?

Reply:
I'm a little unsure about checking `readable` and that causing it to throw. Maybe we could just  return an empty iterator if `!readable` instead of throwing?

Reply:
I also would like to avoid checking for `OutgoingMessage` explicitly.